src/ssl.rs

```diff
@@ -486,6 +563,10 @@ impl<IOCB: IOCallbacks> Session<IOCB> {
             e => unreachable!("{e:?}"),
         }
 
+        // SAFETY: Based on `wolfSSL_key_update_response`][0], required will be populated if the api returns success.
+        // So safety to call `assume_init()` on success case. On error case we panic!
```

Comment:
I got confused here by "On error case we panic", thinking this meant in the match below -- but rereading I think you were talking about the previous match. Can we say "On error we already panicked above" to clarify?

(Aside: I think the use of `e` for the blanket match below is misleading, and contributed to my misreading here, that use of `e` is not "an error" it's "a value other than 0 or 1", IMHO `nr` would be a better name).

Reply:
Yep updated the comment!
But the `e` on blanket match of `assume_init` looks ok to me.
Based on documentation`wolfSSL_key_update_response` api which is supposed to initialize the `required` variable guarantees the value will only be `0` or `1`